controllers/cluster_controller.go

```diff
@@ -138,7 +138,7 @@ func (r *ClusterReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, reterr e
 		r.reconcileMetrics(ctx, cluster)
 
 		// Always attempt to Patch the Cluster object and status after each reconciliation.
-		if err := patchHelper.Patch(ctx, cluster); err != nil {
+		if err := patchHelper.Patch(ctx, cluster, patch.WithStatusObservedGeneration{}); err != nil {
```

Comment:
These (and everywhere else) should check the return error first before adding the patch, something like:

```suggestion

		patchOpts := []patch.Option{}
		if reterr == nil {
				patchOpts = append(patchOpts, patch.WithStatusObservedGeneration{})
		}
		if err := patchHelper.Patch(ctx, cluster, patchOpts); err != nil {
```

Reply:
done